ServerCore/Pages/Puzzles/Edit.cshtml.cs

```diff
@@ -118,12 +118,12 @@ public async Task<IActionResult> OnPostAsync()
 
         public async Task<IActionResult> OnPostAddAuthorAsync()
         {
-            if (!(await _context.EventAuthors.Select(m => m.Author.ID == NewAuthorID && m.Event == Event).AnyAsync()))
+            if (!(await _context.EventAuthors.Where(m => m.Author.ID == NewAuthorID && m.Event == Event).AnyAsync()))
             {
                 return NotFound();
             }
 
-            if (!(await _context.PuzzleAuthors.Select(m => m.PuzzleID == Puzzle.ID && m.AuthorID == NewAuthorID).AnyAsync()))
+            if (!(await _context.PuzzleAuthors.Where(m => m.PuzzleID == Puzzle.ID && m.AuthorID == NewAuthorID).AnyAsync()))
```

Comment:
Rather than use and roundtrip Puzzle.ID, could you just accept puzzleID as a parameter to the method since it's already in the route? That takes care of auth for this too.